### PR TITLE
Feature: Refresh of remote control screen buttons

### PR DESCRIPTION
**Pull request by wutschel**
## Description

This PR is an add-on to https://github.com/xbmc/Official-Kodi-Remote-iOS/pull/311 and refreshes the icons/buttons of the remote control screen, as well as of the remote help screen. Screenshots can be found in #310.

## Summary for release notes

Feature: Refresh remote control screen buttons (aligned with Kodi 19)

### CODE REVIEW

**Comment by wutschel** [comment]
I am adding this PR as work-in-progress to allow testing. It should be added only after the bigger code style and AppDelegate rework PRs to avoid problems rebasing those.

**Comment by wutschel** [comment]
Rebased to master.

**Comment by wutschel** [comment]
Rebased to master.

File: XBMC Remote/RemoteController.m [hunk 1]
```diff
@@ -54,22 +54,7 @@ -(void)hideButton:(NSArray *)buttonsToDo hide:(BOOL)hide{
...
-        transViewY = 66;
-    }
-    else if (transform > 1.0) {
-        // All devices with 320 > width > 414
-        startX = 3;
-        transViewY = 58;
-    }
-        
-    int newWidth = (int) (296 * transform);
+    CGFloat newWidth = remoteControlView.frame.size.width - ANCHORRIGHTPEEK;
```
Comment (kambala-decapitator) [inline XBMC Remote/RemoteController.m:61]: please add underscores to the macro, it's quite difficult to read

File: XBMC Remote/RemoteController.m [hunk 2]
```diff
@@ -98,9 +81,15 @@ - (void)setEmbeddedView{
                            nil]
                     hide: YES];
     }
+    // Place the transitional view in the middle between the two button rows
+    CGFloat lowerButtonUpperBorder = CGRectGetMinY(((UIButton*)[self.view viewWithTag:21]).frame);
+    CGFloat upperButtonLowerBorder = CGRectGetMaxY(((UIButton*)[self.view viewWithTag:6]).frame);
```
Comment (kambala-decapitator) [inline XBMC Remote/RemoteController.m:90]: no need to cast to `UIButton`

File: XBMC Remote/RemoteController.m [hunk 3]
```diff
@@ -128,9 +117,12 @@ - (void)setEmbeddedView{
     [gestureButton addTarget:self action:@selector(toggleGestureZone:) forControlEvents:UIControlEventTouchUpInside];
     
     frame = subsInfoLabel.frame;
-    frame.origin.x = -1 * startX;
-    frame.size.width = newWidth + (startX * 2);
+    frame.origin.x = 0;
+    frame.size.width = newWidth;
     subsInfoLabel.frame = frame;
+    
+    frame = self.view.superview.frame;
```
Comment (kambala-decapitator) [inline XBMC Remote/RemoteController.m:128]: using superview's `frame` (as opposed to `bounds`) in subview's calculations is generally incorrect. What are you trying to achieve here?

File: XBMC Remote/RemoteController.m [hunk 4]
```diff
@@ -143,14 +135,13 @@ - (void)configureView{
         CGRect frame = remoteControlView.frame;
         frame.size.height = frame.size.height *transform;
         frame.size.width = frame.size.width*transform;
-        [remoteControlView setFrame:CGRectMake(frame.origin.x, frame.origin.y, frame.size.width* SCALE_TO_REDUCE_BORDERS, frame.size.height* SCALE_TO_REDUCE_BORDERS)];
+        [remoteControlView setFrame:CGRectMake(frame.origin.x, frame.origin.y, frame.size.width, frame.size.height)];
```
Comment (kambala-decapitator) [inline XBMC Remote/RemoteController.m:142]:
```suggestion
        remoteControlView.frame = frame;
```

File: XBMC Remote/RemoteController.m [hunk 5]
```diff
@@ -161,8 +152,15 @@ - (void)configureView{
          UIViewAutoresizingFlexibleHeight |
          UIViewAutoresizingFlexibleWidth
          ];
-
-        int newHeight = remoteControlView.frame.size.height * newWidth / remoteControlView.frame.size.width;        
+        
+        // Calculate the maximum possible scaling for the remote
+        CGFloat scaleFactorHorizontal = STACKSCROLL_WIDTH / remoteControlView.frame.size.width;
+        CGFloat minViewHeight = MIN(CGRectGetWidth(UIScreen.mainScreen.fixedCoordinateSpace.bounds), CGRectGetHeight(UIScreen.mainScreen.fixedCoordinateSpace.bounds)) - REMOTE_PADDING;
+        CGFloat scaleFactorVertical =  minViewHeight / remoteControlView.frame.size.height;
```
Comment (kambala-decapitator) [inline XBMC Remote/RemoteController.m:163]: please use `CGRectGetHeight` for consistency

File: XBMC Remote/RemoteController.m [hunk 6]
```diff
@@ -161,8 +152,15 @@ - (void)configureView{
...
+        CGFloat transform = MIN(scaleFactorHorizontal, scaleFactorVertical);
+
+        CGFloat newWidth = remoteControlView.frame.size.width * transform;
+        CGFloat newHeight = remoteControlView.frame.size.height * transform;
```
Comment (kambala-decapitator) [inline XBMC Remote/RemoteController.m:167]: `CGRectGet...`

File: XBMC Remote/RemoteController.m [hunk 7]
```diff
@@ -1258,17 +1258,20 @@ - (void)viewDidLoad{
         }
     }
     if ([[UIDevice currentDevice] userInterfaceIdiom] == UIUserInterfaceIdiomPad) {
+        CGRect frame = CGRectMake(self.view.bounds.size.width - 256, self.view.bounds.size.height - TOOLBAR_ICON_SIZE - TOOLBAR_SPACING, TOOLBAR_ICON_SIZE, TOOLBAR_ICON_SIZE);
```
Comment (kambala-decapitator) [inline XBMC Remote/RemoteController.m:1261]: what is 256?

File: XBMC Remote/RemoteController.m [hunk 8]
```diff
@@ -1258,17 +1258,20 @@ - (void)viewDidLoad{
...
         [settingButton setShowsTouchWhenHighlighted:YES];
         [settingButton setImage:[UIImage imageNamed:@"default-right-menu-icon"] forState:UIControlStateNormal];
         settingButton.autoresizingMask = UIViewAutoresizingFlexibleLeftMargin | UIViewAutoresizingFlexibleTopMargin;
         [settingButton addTarget:self action:@selector(addButtonToListIPad:) forControlEvents:UIControlEventTouchUpInside];
+        settingButton.alpha = 0.8;
         [self.view addSubview:settingButton];
         
         UIButton *gestureButton = [UIButton buttonWithType:UIButtonTypeCustom];
-        gestureButton.frame = CGRectMake(self.view.bounds.size.width - 188, self.view.bounds.size.height - 43, 56, 36);
+        frame.origin.x = frame.origin.x + TOOLBAR_PADDING;
```
Comment (kambala-decapitator) [inline XBMC Remote/RemoteController.m:1273]: FYI it's simpler to use `+=` operator:
```suggestion
        frame.origin.x += TOOLBAR_PADDING;
```

File: XBMC Remote/RemoteController.m [hunk 9]
```diff
@@ -1278,24 +1281,25 @@ - (void)viewDidLoad{
         [self.view addSubview:gestureButton];
         
         UIButton *keyboardButton = [UIButton buttonWithType:UIButtonTypeCustom];
-        keyboardButton.frame = CGRectMake(self.view.bounds.size.width - 120, self.view.bounds.size.height - 43, 56, 36);
-        UIImage* keyboardImg = [UIImage imageNamed:@"keyboard_icon"];
+        frame.origin.x = frame.origin.x + TOOLBAR_PADDING;
```
Comment (kambala-decapitator) [inline XBMC Remote/RemoteController.m:1284]: +=

File: XBMC Remote/RemoteController.m [hunk 10]
```diff
@@ -1278,24 +1281,25 @@ - (void)viewDidLoad{
...
-        [keyboardButton setImage:keyboardImg forState:UIControlStateNormal];
+        [keyboardButton setImage:[UIImage imageNamed:@"keyboard_icon"] forState:UIControlStateNormal];
         keyboardButton.autoresizingMask = UIViewAutoresizingFlexibleLeftMargin | UIViewAutoresizingFlexibleTopMargin;
         [keyboardButton addTarget:self action:@selector(toggleVirtualKeyboard:) forControlEvents:UIControlEventTouchUpInside];
         keyboardButton.alpha = 0.8;
         [self.view addSubview:keyboardButton];
 
-        UIButton *helpButton = [UIButton buttonWithType:UIButtonTypeInfoLight];
+        UIButton *helpButton = [UIButton buttonWithType:UIButtonTypeCustom];
+        frame.origin.x = frame.origin.x + TOOLBAR_PADDING;
```
Comment (kambala-decapitator) [inline XBMC Remote/RemoteController.m:1295]: +=

**Review by kambala-decapitator — CHANGES_REQUESTED** [review]
> Only keep remote images with 2x resolution

are you sure that's correct? We support iOS 9 which iPad 2 is capable of running, and it's non-retina.

**Comment by wutschel** [comment]
> are you sure that's correct? We support iOS 9 which iPad 2 is capable of running, and it's non-retina.

I was wondering if we really need the x1 images as there are also no 1x images for the gesture zone. Of course I can easily add the 1x images for remote help. Should I?

File: XBMC Remote/RemoteController.m (hunk 1 above)
Comment (wutschel) [reply XBMC Remote/RemoteController.m:61]: You mean `ANCHORRIGHTPEEK`? If so, would you mind if I am doing this in a separate coding styles PR? This change touches several other files which are not connected to this PR.

File: XBMC Remote/RemoteController.m (hunk 2 above)
Comment (wutschel) [reply XBMC Remote/RemoteController.m:90]: Ok, will be done with next uodate.

File: XBMC Remote/RemoteController.m (hunk 4 above)
Comment (wutschel) [reply XBMC Remote/RemoteController.m:142]: Ok, will be part of next update.

File: XBMC Remote/RemoteController.m (hunk 10 above)
Comment (wutschel) [reply XBMC Remote/RemoteController.m:1295]: Ok, will do this the other 3 occasions as well.

File: XBMC Remote/RemoteController.m (hunk 7 above)
Comment (wutschel) [reply XBMC Remote/RemoteController.m:1261]: Argh, I already defined `TOOLBAR_START` for this. It also has the correct value, 256 was even a bit off. Will be part of next update.

File: XBMC Remote/RemoteController.m (hunk 5 above)
Comment (wutschel) [reply XBMC Remote/RemoteController.m:163]: No problem, done with next update.

File: XBMC Remote/RemoteController.m (hunk 6 above)
Comment (wutschel) [reply XBMC Remote/RemoteController.m:167]: No problem, done with next update.

File: XBMC Remote/RemoteController.m (hunk 1 above)
Comment (kambala-decapitator) [reply XBMC Remote/RemoteController.m:61]: sure that's ok

**Comment by kambala-decapitator** [comment]
> > are you sure that's correct? We support iOS 9 which iPad 2 is capable of running, and it's non-retina.
> 
> I was wondering if we really need the x1 images as there are also no 1x images for the gesture zone. Of course I can easily add the 1x images for remote help. Should I?

yes please

File: XBMC Remote/RemoteController.m (hunk 3 above)
Comment (wutschel) [reply XBMC Remote/RemoteController.m:128]: I wanted to draw the background image ("remote_temp") with same relative dimensions for this use case (embedded view) as for the standard remote view. This background image is at the same hierarchy level as the remote control view itself. 
The embedded view is drawn into a cell of a height of only 50 (see call of `[remoteControllerView setEmbeddedView]` in `RightMenuViewController.m`). The same dimensions I get when using `self.view.superview.frame` inside `setEmbeddedView`.

**Comment by wutschel** [comment]
Rebased to master and updated regarding the review comments, except 

- `ANCHORRIGHTPEEK`: Will be done in separate PR
- Using superview's `frame` in `setEmbeddedView `: Still open

File: XBMC Remote/RemoteController.m (hunk 3 above)
Comment (wutschel) [reply XBMC Remote/RemoteController.m:128]: I get the same desired behaviour if I remove

```
frame = self.view.superview.frame;
panFallbackImageView.frame = frame;
```

and set the frame after `setEmbeddedView` in `RightMenuViewController.m` like this:

```
[remoteControllerView setEmbeddedView];
remoteControllerView.panFallbackImageView.frame = cell.frame;
```

Comment (kambala-decapitator) [reply XBMC Remote/RemoteController.m:128]: the latter looks much clearer, please use that

Comment (wutschel) [reply XBMC Remote/RemoteController.m]: Done.

**Comment by wutschel** [comment]
Resolved last review comment regarding `setEmbeddedView`.

**Pull request merged**